File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.py
'''
매일 1명의 가수가 노래, 시청자가 점수 부여
지금까지 점수들 중 상위 k번째 이내이면 명예의 전당으로
명예의 전당에는 k명만 있어야 함
score => 사람들의 점수 
'''
def solution(k, score):
    answer = []
    result = []
    for i in range(len(score)):
        answer.append(score[i]) # i를 일단 넣어
        answer.sort(reverse=True) # 정렬해 거꾸로
        answer = answer[:k] # k등까지 자르기
        result.append(answer[-1])
    return result

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.ArrayList;
import java.util.Collections;
class Solution {
    public ArrayList<Integer> solution(int k, int[] score) {
        ArrayList<Integer> answer = new ArrayList<>();
		ArrayList<Integer> result = new ArrayList<>();
        
        for(int i = 0; i < score.length; i++) {
        	answer.add(score[i]);
        	Collections.sort(answer,Collections.reverseOrder());
        	if(answer.size() >= k) result.add(answer.get(k-1));
        	else if (answer.size() < k) result.add(answer.get(answer.size()-1));
        }
        
        return result;
    }
}